compiler/executable/match_/planner/vertex/variable.rs

```diff
@@ -271,82 +308,59 @@ impl ThingPlanner {
             ordered.iter().filter_map(VertexId::as_pattern_id).any(|id| adjacent.contains(&id))
         }
     }
-}
 
-impl Costed for ThingPlanner {
-    fn cost(&self, inputs: &[VertexId], graph: &Graph<'_>) -> ElementCost {
-        let bounds = chain!(&self.bound_value_equal, &self.bound_value_above, &self.bound_value_below).collect_vec();
-        for &i in inputs {
-            let VertexId::Variable(i) = i else { continue };
-            if !bounds.contains(&&Input::Variable(i)) {
-                return ElementCost::FREE;
+    fn selectivity(&self) -> f64 {
```

Comment:
Why does this not take the ordered prefix? Here, if you query `match $x has id $x-id; $y has id $y-id; $x-id == $y-id;`, both `$x-id` and `$y-id` will be highly selective, but only the later one in execution can actually make use of that.

Reply:
Good point

Reply:
Done!